funnel/views/index.py

```diff
@@ -17,8 +18,14 @@ def index_jsonify(data):
 def index():
     g.profile = None
     g.permissions = []
+    today = datetime.utcnow().date()
     projects = Project.fetch_sorted(legacy=False).all()  # NOQA
-    return {'projects': projects}
+    past_projects = [project for project in projects if project.state.PUBLISHED and project.date_upto < today]
+    all_projects = [project for project in projects if project.state.PUBLISHED and project.date_upto >= today]
+    upcoming_projects = [project for project in projects if project.schedule_state.PUBLISHED and project.date_upto >= today][:3]
+    open_cfp_projects = [project for project in projects if project.cfp_state.OPEN and project.date_upto >= today]
+    return {'projects': projects, 'past_projects': past_projects, 'all_projects': all_projects,
+        'upcoming_projects': upcoming_projects, 'open_cfp_projects': open_cfp_projects}
```

Comment:
Make all of these conditional states, and use StateManager's ability to group. https://github.com/hasgeek/coaster/blob/master/coaster/sqlalchemy/statemanager.py#L878